File: 01jupyter/nb/source/01_intro/include/func.cpp
#include <cstdio>

/* 2数の平均を返す関数 (引数2つ, 返り値1つ) */
double average(double a, double b) {
  return (a + b) / 2.0;
}

/* n! を計算する関数 */
long factorial(int n) {
  long p = 1;
  for (int i = 2; i <= n; i++) {
    p = p * i;
  }
  return p;
}

int main() {
  printf("average(3.0, 4.0) = %f\n", average(3.0, 4.0));
  printf("factorial(5)      = %ld\n", factorial(5));
  return 0;
}

File: 01jupyter/nb/source/01_intro/include/func.f90
module func_mod
contains
  ! 2数の平均を返す関数 (引数2つ, 返り値1つ)
  function average(a, b) result(m)
    real(8), intent(in) :: a, b
    real(8) :: m
    m = (a + b) / 2.0d0
  end function average

  ! n! を計算する関数
  function factorial(n) result(p)
    integer, intent(in) :: n
    integer(8) :: p
    integer :: i
    p = 1
    do i = 2, n
       p = p * i
    end do
  end function factorial
end module func_mod

program func
  use func_mod
  print "(a,f0.6)", "average(3.0, 4.0) = ", average(3.0d0, 4.0d0)
  print "(a,i0)",   "factorial(5)      = ", factorial(5)
end program func
